File: sirius-ruby-client-web/lib/sirius/web/classes/web_radio_button.rb
module Sirius
  module Client
    module Web
      module Classes
        class WebRadioButton < WebButton
          def initialize(parent, locator, parent_element = nil)
            super(parent, locator, parent_element)
          end

          def checked?
            client.core.isSelected(@parent_element, @locator)
          end

          def check
            click unless checked?
          end
        end
      end
    end
  end
end

File: SiriusCSharp.Client/Sirius.CSharp.Client.Web/Classes/WebRadioButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Sirius.CSharp.Client.Web.Classes
{
    public class WebRadioButton : WebButton
    {
        public WebRadioButton(Frame parent, String parentElement, String locator)
            : base(parent, parentElement, locator)
        {
            ;
        }

        public WebRadioButton(Frame parent, String locator)
            : base(parent, locator)
        {
            ;
        }

        public bool isChecked()
        {
            return this.client().isSelected(parentElement, locator);
        }

        public void check()
        {
            if (!isChecked())
            {
                click();
            }
        }
    }
}
